pkg/services/sqlstore/transactions_test.go

```diff
@@ -65,3 +68,49 @@ func TestIntegrationReuseSessionWithTransaction(t *testing.T) {
 		}))
 	})
 }
+
+func TestIntegrationPublishAfterCommitWithNestedTransactions(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ss := InitTestDB(t)
+	ctx := context.Background()
+
+	// On X success
+	var xHasSucceeded bool
+	ss.Bus().AddEventListener(func(ctx context.Context, e *X) error {
+		xHasSucceeded = true
+		fmt.Printf("Succeeded and committed: %T\n", e)
+		return nil
+	})
+
+	// On Y success
+	var yHasSucceeded bool
+	ss.Bus().AddEventListener(func(ctx context.Context, e *Y) error {
+		yHasSucceeded = true
+		fmt.Printf("Succeeded and committed: %T\n", e)
+		return nil
+	})
+
+	err := ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
+		fmt.Println("Outer transaction: doing X... success!")
+		sess.PublishAfterCommit(&X{})
+
+		ss.InTransaction(ctx, func(ctx context.Context) error {
+			fmt.Println("Inner transaction: doing Y... success!")
+			sess.PublishAfterCommit(&Y{})
+			return nil
+		})
```

Comment:
This reuses the `sess` from the outer function, I don't think that's correct?

Shouldn't it be like this?
```suggestion
		ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
			fmt.Println("Inner transaction: doing Y... success!")
			sess.PublishAfterCommit(&Y{})
			return nil
		})
```

Reply:
I'm not sure, honestly, I sometimes get confused with transactions and sessions, etc I have to re-read the implementation details (my fault!) haha, so yeah, I feel more comfortable with going with what you suggested!

In fact, guess it could even make sense to test more than one use case (e.g. verify that with explicitly different sessions, both transactions are independent). But yeah, let's correct this one, commit and iterate.

Thanks for your help! 😊